unit/unit.go

```diff
@@ -169,6 +170,19 @@ func (h *Hash) Empty() bool {
 	return *h == Hash{}
 }
 
+func HashFromHexString(key string) (Hash, error) {
+	h := Hash{}
+	out, err := hex.DecodeString(key)
+	if err != nil {
+		return h, err
+	}
+	if len(out) != sha1.Size {
+		return h, fmt.Errorf("Size of %d longer than SHA1 size of %d", len(out), sha1.Size)
+	}
+	copy(h[:], out[0:sha1.Size])
```

Comment:
I think `h[:]` can just be `h`, `out[0:sha1.Size]` can be `out[:sha1.Size]` for sure.